### PR TITLE
Allow ngrams in URL text.

### PR DESCRIPTION
**Pull request by mhilton**
This sorts out a number of search issues:
- ngrams match in URLs
- Bundle Tags are supported in search
- Support for charms and bundles with revision 0

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results: http://ci.jujugui.org:8080/job/charmstore/451/

File: internal/storetesting/essettings.go [hunk 1]
```diff
@@ -3,288 +3,33 @@
...
+)
 
 var esIndex map[string]interface{}
 var esMapping map[string]interface{}
 
 func init() {
-	if err := json.Unmarshal(esIndexText, &esIndex); err != nil {
+	path := filepath.Join(os.Getenv("GOPATH"), "src", "github.com", "juju", "charmstore", "internal", "elasticsearch", "definitions")
+	f, err := os.Open(filepath.Join(path, "index.json"))
+	if err != nil {
```
Comment (urosj) [inline internal/storetesting/essettings.go:18]: Nice! :)

**Comment by urosj** [comment]
:+1:

**Comment by mhilton** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Pull request merged**